server/aesdsocket.c

```diff
@@ -209,10 +341,36 @@ void myCustomExit(){
   }
 
   if(signalCaught){
-    remove(SOCKET_PATH);                                                                           // Remove temp file from file system //
+    unlink(SOCKET_PATH);                                                                           // Remove temp file from file system //
+  }
+
+  int returnStatus;
+  slist_data_t *tNode, *t_node_temp;
+  SLIST_FOREACH_SAFE(tNode, &head, entries, t_node_temp){
+    if(tNode->threadParam.threadComplete){                                                         // Check if thread complete //
```

Comment:
@Vignesh-Vadivel in this case you won't want to require threadComplete - you'll want to join all threads, waiting for them to exit if necessary.